app/services/publish.js

```diff
@@ -0,0 +1,106 @@
+import Service, { inject as service } from '@ember/service';
+import { task, all } from 'ember-concurrency';
+import { tracked } from '@glimmer/tracking';
+
+export class Extract {
+  @tracked treatmentId;
+  @tracked position;
+  @tracked document;
+  @tracked errors;
+
+  constructor(treatmentId, position, document, errors) {
+    this.treatmentId = treatmentId;
+    this.position = position;
+    this.document = document;
+    this.errors = errors;
+  }
+}
+
+export default class PublishService extends Service {
+  @tracked treatmentExtractsMap;
+  @service store;
+
+  constructor(...args) {
+    super(...args);
+    this.treatmentExtractsMap = new Map();
+  }
+
+  get treatmentExtracts() {
+    return [...this.treatmentExtractsMap.values()].sortBy('position');
+  }
+
+  get isIdle() {
+    return this.loadExtractsTask.isIdle;
+  }
+
+  get loadExtractsTask() {
+    return this._loadExtractsTask.unlinked();
```

Comment:
I'll have to read up on what unlinked means here

Reply:
it is the stupidest thing about ember-concurrency
https://github.com/machty/ember-concurrency/issues/161
https://github.com/machty/ember-concurrency/issues/249
You get a warning in the console at least. But it is really annoying, especially since you can pass an unlinked task through like here, but if it needs to be linked, you can't do that since you have to do it right before a perform or something